### ISSUE TITLE
Tried to load unspecified class: Symbol when migrating move_user_settings.rb

### ISSUE THREAD

**Issue by tcitworld** [glitch-soc/mastodon#2162]
### Steps to reproduce the problem

1. Get latest commit (02d5c69)
2. `rails db:migrate`

```
Migrating to MoveUserSettings (20230215074423)
== 20230215074423 MoveUserSettings: migrating =================================
rails aborted!
StandardError: An error has occurred, this and all later migrations canceled:

Tried to load unspecified class: Symbol
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:55:in `value'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:66:in `block (2 levels) in up'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:65:in `each'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:65:in `block in up'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:71:in `block (2 levels) in find_each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:71:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:71:in `block in find_each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:138:in `block in find_in_batches'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:245:in `block in in_batches'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:229:in `loop'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:229:in `in_batches'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:137:in `find_in_batches'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:70:in `find_each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/querying.rb:22:in `find_each'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:60:in `up'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:870:in `public_send'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:870:in `exec_migration'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:851:in `block (2 levels) in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:850:in `block in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:462:in `with_connection'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:849:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/strong_migrations-0.8.0/lib/strong_migrations/migration.rb:5:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/railtie.rb:28:in `block in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/strategy.rb:59:in `wrap'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy.rb:154:in `strategy'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/railtie.rb:28:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1037:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1329:in `block in execute_migration_in_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1380:in `block in ddl_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `block in transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/transaction.rb:319:in `block in within_new_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/transaction.rb:317:in `within_new_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/transactions.rb:209:in `transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1380:in `ddl_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1328:in `execute_migration_in_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1302:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1302:in `migrate_without_lock'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1251:in `block in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1401:in `block in with_advisory_lock'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1416:in `block in with_advisory_lock_connection'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:462:in `with_connection'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1416:in `with_advisory_lock_connection'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1397:in `with_advisory_lock'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1251:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/railtie.rb:28:in `block in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/strategy.rb:59:in `wrap'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy.rb:154:in `strategy'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/railtie.rb:28:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1086:in `up'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1061:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/tasks/database_tasks.rb:237:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/strong_migrations-0.8.0/lib/strong_migrations/database_tasks.rb:5:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/railties/databases.rake:92:in `block (3 levels) in <top (required)>'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/railties/databases.rake:90:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/railties/databases.rake:90:in `block (2 levels) in <top (required)>'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `synchronize'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/commands/rake/rake_command.rb:24:in `block (2 levels) in perform'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/commands/rake/rake_command.rb:24:in `block in perform'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/rake_module.rb:59:in `with_application'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/commands/rake/rake_command.rb:18:in `perform'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/command.rb:50:in `invoke'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/commands.rb:18:in `<top (required)>'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'

Caused by:
Psych::DisallowedClass: Tried to load unspecified class: Symbol
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:55:in `value'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:66:in `block (2 levels) in up'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:65:in `each'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:65:in `block in up'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:71:in `block (2 levels) in find_each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:71:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:71:in `block in find_each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:138:in `block in find_in_batches'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:245:in `block in in_batches'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:229:in `loop'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:229:in `in_batches'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:137:in `find_in_batches'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/relation/batches.rb:70:in `find_each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/querying.rb:22:in `find_each'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:60:in `up'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:870:in `public_send'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:870:in `exec_migration'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:851:in `block (2 levels) in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:850:in `block in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:462:in `with_connection'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:849:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/strong_migrations-0.8.0/lib/strong_migrations/migration.rb:5:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/railtie.rb:28:in `block in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/strategy.rb:59:in `wrap'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy.rb:154:in `strategy'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/railtie.rb:28:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1037:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1329:in `block in execute_migration_in_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1380:in `block in ddl_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `block in transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/transaction.rb:319:in `block in within_new_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:26:in `block (2 levels) in synchronize'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activesupport-6.1.7.3/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/transaction.rb:317:in `within_new_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/database_statements.rb:320:in `transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/transactions.rb:209:in `transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1380:in `ddl_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1328:in `execute_migration_in_transaction'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1302:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1302:in `migrate_without_lock'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1251:in `block in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1401:in `block in with_advisory_lock'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1416:in `block in with_advisory_lock_connection'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/connection_adapters/abstract/connection_pool.rb:462:in `with_connection'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1416:in `with_advisory_lock_connection'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1397:in `with_advisory_lock'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1251:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/railtie.rb:28:in `block in migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/strategy.rb:59:in `wrap'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy.rb:154:in `strategy'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/chewy-7.2.7/lib/chewy/railtie.rb:28:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1086:in `up'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/migration.rb:1061:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/tasks/database_tasks.rb:237:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/strong_migrations-0.8.0/lib/strong_migrations/database_tasks.rb:5:in `migrate'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/railties/databases.rake:92:in `block (3 levels) in <top (required)>'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/railties/databases.rake:90:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/activerecord-6.1.7.3/lib/active_record/railties/databases.rake:90:in `block (2 levels) in <top (required)>'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `synchronize'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/commands/rake/rake_command.rb:24:in `block (2 levels) in perform'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/commands/rake/rake_command.rb:24:in `block in perform'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/rake-13.0.6/lib/rake/rake_module.rb:59:in `with_application'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/commands/rake/rake_command.rb:18:in `perform'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/command.rb:50:in `invoke'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/railties-6.1.7.3/lib/rails/commands.rb:18:in `<top (required)>'
/home/mastodon/live/vendor/bundle/ruby/3.2.0/gems/bootsnap-1.16.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```

### Expected behaviour

Migration success

### Actual behaviour

Failure to run migration MoveUserSettings (20230215074423)

### Specifications

Glitch-soc : `02d5c69`
Ruby: `3.2.2`

**Comment by ClearlyClaire** [comment]
Can you apply this patch and run the migration again? It should fail again but give me more insight on what is going on:

```patch
diff --git a/db/migrate/20230215074423_move_user_settings.rb b/db/migrate/20230215074423_move_user_settings.rb
index 351a8b61df..223a9e8d00 100644
--- a/db/migrate/20230215074423_move_user_settings.rb
+++ b/db/migrate/20230215074423_move_user_settings.rb
@@ -53,6 +53,8 @@ class MoveUserSettings < ActiveRecord::Migration[6.1]
 
     def value
       YAML.safe_load(self[:value], permitted_classes: [ActiveSupport::HashWithIndifferentAccess]) if self[:value].present?
+    rescue
+      raise "Error while loading #{self[:var]}: #{self[:value]}"
     end
   end
 
```

**Comment by tcitworld** [comment]
Thanks, this give the culprit:
```
Error while loading notification_emails: ---
:follow: false
:follow_request: true
:reblog: false
:favourite: false
:mention: false
:digest: true
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:57:in `rescue in value'
/home/mastodon/live/db/migrate/20230215074423_move_user_settings.rb:54:in `value'
```

**Comment by ClearlyClaire** [comment]
oh, the hash's keys are symbols… ok

Can you try the following patch?

```patch
diff --git a/db/migrate/20230215074423_move_user_settings.rb b/db/migrate/20230215074423_move_user_settings.rb
index 351a8b61df..8dbca9ecd2 100644
--- a/db/migrate/20230215074423_move_user_settings.rb
+++ b/db/migrate/20230215074423_move_user_settings.rb
@@ -52,7 +52,7 @@ class MoveUserSettings < ActiveRecord::Migration[6.1]
     end
 
     def value
-      YAML.safe_load(self[:value], permitted_classes: [ActiveSupport::HashWithIndifferentAccess]) if self[:value].present?
+      YAML.safe_load(self[:value], permitted_classes: [ActiveSupport::HashWithIndifferentAccess, Symbol]) if self[:value].present?
     end
   end
 
```

**Comment by tcitworld** [comment]
The migration passes. `MoveGlitchUserSettings` chokes on the same hash and the above patch fixes it as well.

Thanks!

**Comment by ClearlyClaire** [comment]
Reopening the issue since it's likely a real code issue (shared by upstream too, but the `MoveGlitchUserSettings` is glitch-specific)

**Comment by meatballhat** [comment]
fwiw we hit the same problem during a recent [upgrade attempt](https://toot.cat/@meatballhat/110152216164828638) and saw exactly two settings that included `Symbol`:

- `notification_emails`
- `interactions`

<details>
  <summary>rails console script bits</summary>

```ruby
def find_sad_settings
  sad_settings = []

  Setting.find_each do |s|
    next if s[:thing_type] != 'User'
    YAML.safe_load(s[:value], permitted_classes: [ActiveSupport::HashWithIndifferentAccess]) if s[:value].present?
    nil
  rescue => e
    sad_settings << s
    puts "failed to yaml load #{s[:var]} (thing_id=#{s[:thing_id]}): #{e}"
  end

  sad_settings
end
```

```irb
irb(main):002:0> find_sad_settings.map { |s| s[:var] }.uniq
=> ["notification_emails", "interactions"]
```

</details>

**Comment by ClearlyClaire** [comment]
This should be fixed once #2170 gets merged
